src/js/utils/styles.js

```diff
@@ -208,18 +234,56 @@ export const focusStyle = css`
   }
 `;
 
+export const getInputPadBySide = (props, side) => {
+  if (typeof props.theme.global.input.padding !== 'object') {
+    return props.theme.global.input.padding;
+  }
+
+  let orientation;
+  if (side === 'left' || side === 'right') orientation = 'horizontal';
+  else if (side === 'top' || side === 'bottom') orientation = 'vertical';
+  else orientation = undefined;
+
+  // if individual side isn't available, fallback to the
+  // orientation if possible
+  const pad =
+    props.theme.global.input.padding[side] ||
+    props.theme.global.input.padding[orientation];
+
+  // for backwards compatibility we need to add back the control border width
+  // in each case, check first if the value is an edgeSize,
+  // then if it's a string value.
+  const adjustedPad =
+    parseMetricToNum(`${props.theme.global.edgeSize[pad] || pad}px`) +
+    parseMetricToNum(`${props.theme.global.control.border.width}px`);
```

Comment:
I thought we weren't going to do anything with border if the `theme.global.input.padding` is an object?

Reply:
This is used on StyledPlaceholder in TextInput and MaskedInput. Based on how grommet was functioning prior to these changes, the padding was subtracting the border width from the theme value, but the placholder was not. So, for example, if the value defined in the theme for `global.input.control.padding` was `12px`, then the padding value would be calculated to `11px`, but the styledplaceholder pad would still be `12px`. Because we're now placing the subtraction into the theme itself, we have to add back in the border width.

Unless, we don't want these values to be different, but for backwards compatibility this addition needs to remain.

Reply:
Ah, thanks for the details. I think it would help to add some of this as a comment above the function here, mentioning that it is for placeholder and icon placement with TextInput and MaskedInput.

Reply:
Sounds good, let me add that comment in.